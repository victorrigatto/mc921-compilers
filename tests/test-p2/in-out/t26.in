char v[] = {'a', 'b', 'c', 'd'};
int main(){
    for (int i = 1; i < 5; i=i+1)
        print(v[i]);
        break;
    return 0;
}
