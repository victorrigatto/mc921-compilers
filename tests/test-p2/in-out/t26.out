Program:
    GlobalDecl:
        Decl: ID(name=v)
            ArrayDecl:
                VarDecl:
                    Type: char @ 1:1
            InitList: @ 1:13
                Constant: char, 'a' @ 1:13
                Constant: char, 'b' @ 1:18
                Constant: char, 'c' @ 1:23
                Constant: char, 'd' @ 1:28
    FuncDef:
        Type: int @ 2:1
        Decl: ID(name=main)
            FuncDecl:
                VarDecl:
                    Type: int @ 2:1
        Compound: @ 2:11
            For: @ 3:5
                DeclList: @ 3:5
                    Decl: ID(name=i)
                        VarDecl:
                            Type: int @ 3:10
                        Constant: int, 1 @ 3:18
                BinaryOp: < @ 3:21
                    ID: i @ 3:21
                    Constant: int, 5 @ 3:25
                Assignment: = @ 3:28
                    ID: i @ 3:28
                    BinaryOp: + @ 3:30
                        ID: i @ 3:30
                        Constant: int, 1 @ 3:32
                Print: @ 4:9
                    ArrayRef: @ 4:15
                        ID: v @ 4:15
                        ID: i @ 4:17
            Break: @ 5:9
            Return: @ 6:5
                Constant: int, 0 @ 6:12
